packages/fixtures/src/serverless-tester.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  APIGatewayProxyEvent,
+  APIGatewayProxyResult,
+  Context,
+} from 'aws-lambda';
+import crypto from 'crypto';
+import serverless, { Application, Handler } from 'serverless-http';
+
+export default class ServerlessTester {
+  private handler: Handler;
+
+  private readonly signingSecret: string;
+
+  constructor(app: Application, signingSecret: string) {
+    this.handler = serverless(app);
+    this.signingSecret = signingSecret;
+  }
+
+  private createSlackRequest(
+    data: any,
+    path = '/slack/events',
+  ): Partial<APIGatewayProxyEvent> {
+    const body = JSON.stringify(data);
+    const version = 'v0';
+    const timestamp = Math.floor(Date.now() / 1000);
+    const hmac = crypto.createHmac('sha256', this.signingSecret);
+
+    hmac.update(`${version}:${timestamp}:${body}`);
+
+    return {
+      body,
+      path,
+      headers: {
+        'content-type': 'application/json',
+        'x-slack-request-timestamp': timestamp.toString(),
+        'x-slack-signature': `${version}=${hmac.digest('hex')}`,
+      },
+      httpMethod: 'POST',
+    };
+  }
+
+  public sendHttp(
+    event: Partial<APIGatewayProxyEvent>,
+  ): Promise<APIGatewayProxyResult> {
+    return this.handler(event as APIGatewayProxyEvent, {} as Context);
+  }
+
+  public sendSlackEvent(
+    event: Record<string, any>,
+    path?: string,
+  ): Promise<APIGatewayProxyResult> {
+    return this.sendHttp(this.createSlackRequest(event, path));
+  }
```

Comment:
this is basically just a compose/pipe wrapper? chaining functions like this hurts a bit. could rewrite a lot of this file with ramda to be a bit cleaner/point free.

something like

```ts
  public sendSlackEvent: (
    event: Record<string, any>,
    path?: string,
  ) => Promise<APIGatewayProxyResult> = pipe(
    this.createSlackRequest,
    this.sendHttp,
  );
```

maybe a future thing.

Reply:
Yeah, ramda hasn't quite made it into my toolchain yet. I'm still waiting for you to start going wild with it so patterns become obvious to me.